packages/agent/src/agent/http/index.ts

```diff
@@ -424,64 +435,172 @@ export class HttpAgent implements Agent {
     fields: QueryFields,
     identity?: Identity | Promise<Identity>,
   ): Promise<ApiQueryResponse> {
-    const id = await (identity !== undefined ? await identity : await this._identity);
-    if (!id) {
-      throw new IdentityInvalidError(
-        "This identity has expired due this application's security policy. Please refresh your authentication.",
-      );
-    }
-
-    const canister = typeof canisterId === 'string' ? Principal.fromText(canisterId) : canisterId;
-    const sender = id?.getPrincipal() || Principal.anonymous();
+    const makeQuery = async () => {
+      const id = await (identity !== undefined ? await identity : await this._identity);
+      if (!id) {
+        throw new IdentityInvalidError(
+          "This identity has expired due this application's security policy. Please refresh your authentication.",
+        );
+      }
 
-    const request: QueryRequest = {
-      request_type: ReadRequestType.Query,
-      canister_id: canister,
-      method_name: fields.methodName,
-      arg: fields.arg,
-      sender,
-      ingress_expiry: new Expiry(DEFAULT_INGRESS_EXPIRY_DELTA_IN_MSECS),
-    };
+      const canister = typeof canisterId === 'string' ? Principal.fromText(canisterId) : canisterId;
```

Comment:
It looks like Principal.from can create a principal from string or principal.
Could this line be replaced with `Principal.from(canisterId);` to simplify it?

Reply:
It can
```suggestion
      const canister = Principal.from(canisterId);
```